packages/bitcore-wallet-client/src/lib/common/utils.ts

```diff
@@ -505,7 +505,7 @@ export class Utils {
             ...recipients[index],
             tag: destinationTag ? Number(destinationTag) : undefined,
             chain: _chain,
-            nonce: Number(txp.nonce) + Number(index),
+            nonce: txp.nonce ? Number(txp.nonce) + Number(index): null,
```

Comment:
This and above should be `txp.nonce != null ? ...` so that if txp.nonce == 0 it doesn't set it to null.